ignore/src/dir.rs

```diff
@@ -109,8 +109,11 @@ struct IgnoreInner {
     /// The absolute base path of this matcher. Populated only if parent
     /// directories are added.
     absolute_base: Option<Arc<PathBuf>>,
-    /// Explicit ignore matchers specified by the caller.
+    /// Explicit global ignore matchers specified by the caller.
     explicit_ignores: Arc<Vec<Gitignore>>,
+    /// Ignore files used in addition to `.ignore`
+    custom_ignore_filenames: Vec<OsString>,
```

Comment:
This should be in a `Arc<Vec<OsString>>`. Otherwise you're going to wind up copying it everywhere.